src/main/java/com/cburch/logisim/circuit/appear/DefaultEvolutionAppearance.java

```diff
@@ -142,34 +146,50 @@ private static void placePins(
               : Wire.WIDTH >> 1;
       final var height =
           (pin.getAttributeValue(StdAttr.WIDTH).getWidth() > 1) ? Wire.WIDTH_BUS : Wire.WIDTH;
-      Rectangle rect;
       if (isLeftSide) {
         ldX = 15;
         hAlign = EditableLabel.LEFT;
-        rect = new Rectangle(x, y - offset, 10, height);
       } else {
         ldX = -15;
         hAlign = EditableLabel.RIGHT;
+      }
+      if (pin.getAttributeSet().containsAttribute(StdAttr.LABEL)) {
+        var label = pin.getAttributeValue(StdAttr.LABEL);
+        if (label.equals("clk")) {
+          y = bottomY;
```

Comment:
Updating `y` here is why I moved the rect and `AppearancePort` to be drawn after the (label OR clk symbol) part.